chore(gatsby-parcel-config): stabilize ts compilation parcel output

pieh opened the pull request:
## Description

Right now, because of https://github.com/parcel-bundler/parcel/issues/5476#issuecomment-769058504 when you have `.ts` file only in local plugin, but not in the root of the repo the output is "messed up" - compiled files are in unexpected locations which means gatsby can't load proper gatsby-node file.

This PR looks to first add failing test case covering above and in next commit add "workoarund" (by adding customer namer plugin)

LekoArts commented:
Do we not want to move this into our monorepo? Also Parcel wants us to use `@gatsbyjs` scope: https://parceljs.org/plugin-system/authoring-plugins/#naming

pieh commented:
> Do we not want to move this into our monorepo? Also Parcel wants us to use @gatsbyjs scope: [parceljs.org/plugin-system/authoring-plugins/#naming](https://parceljs.org/plugin-system/authoring-plugins/#naming)

For now I renamed package to be `@gatsbyjs/` scoped and published and adjusted dependencies in this PR. Adding the package itself to monorepo would be follow up
